src/search_procedure.jl

```diff
@@ -18,3 +18,18 @@ function search(g::Grammar, problem::Problem, depth::Int, start::Symbol, enumera
     end
     return nothing
 end
+
+function search_it(g::Grammar, problem::Problem, iterator)::Any
```

Comment:
This looks like a rather important function, so a docstringc would be nice.
I think `it` could be of type `ExpressionIterator`